static/elements/chromedash-dialog.js

```diff
@@ -0,0 +1,159 @@
+import {LitElement, css, html} from 'lit-element';
+import '@polymer/iron-icon';
+
+// This is a simplfied version of chops-dialog:
+// https://source.chromium.org/chromium/infra/infra/+/main:appengine/monorail/static_src/elements/chops/chops-dialog/chops-dialog.js
+
+class ChromedashDialog extends LitElement {
+  static get properties() {
+    return {
+      opened: {type: Boolean, reflect: true},
+    };
+  }
+
+  static get styles() {
+    return css`
+      :host {
+        position: fixed;
+        z-index: 9999;
+        left: 0;
+        top: 0;
+        width: 100%;
+        height: 100%;
+        overflow: auto;
+        background-color: rgba(0,0,0,0.4);
+        display: flex;
+        align-items: center;
+        justify-content: center;
+      }
+      :host(:not([opened])), [hidden] {
+        display: none;
+        visibility: hidden;
+      }
+      :host([closeOnOutsideClick]),
+      :host([closeOnOutsideClick]) .dialog::backdrop {
+        /* TODO(zhangtiff): Deprecate custom backdrop in favor of native
+        * browser backdrop.
+        */
+        cursor: pointer;
+      }
+      .dialog {
+        background: none;
+        border: 0;
+        max-width: 90%;
+      }
+      .dialog-content {
+        /* This extra div is here because otherwise the browser can't
+        * differentiate between a click event that hits the dialog element or
+        * its backdrop pseudoelement.
+        */
+        box-sizing: border-box;
+        background: white;
+        padding: 1em 16px;
+        cursor: default;
+        box-shadow: 0px 3px 20px 0px hsla(0, 0%, 0%, 0.4);
+        width: var(--dialog-width-param);
+        max-width: var(--dialog-max-width-param, 100%);
+      }
+      #close-icon {
+        background: transparent;
+        border: 0;
+        float: right;
+        cursor: pointer;
+      }
+
+    `;
+  }
+
+  render() {
```

Comment:
very nit: I think normally methods in components are ordered:

- static
- constructor
- lifecycles
- other methods
- render

Reply:
Reordered.  I had gotten the other ordering from chops-dialog, but it makes sense to make it follow the pattern you describe like our other code.